utils/env_preferences/src/error.rs

```diff
@@ -0,0 +1,66 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::{
+    ffi::{FromVecWithNulError, IntoStringError},
+    str::Utf8Error,
+};
+
+#[derive(Debug)]
+pub enum RetrievalError {
+    /// Error converting from `CString` to `String`
+    CStringConversionError(IntoStringError),
```

Comment:
`IntoStringError` contains the `CString` and a `Utf8Error`. We only care about the UTF-8 error, we don't want to propagate the invalid `CString` to the caller.